lib/enumerables.rb

```diff
@@ -0,0 +1,230 @@
+  def my_each(&block)
+    arr = self
+    arr = arr.to_a
+    count = 0
+    if arr.is_a?(Hash)
+      while count < arr.length
+        yield arr[i][0], arr[i][1]
+        count += 1
+      end
+      self
+    elsif block
+      while count < arr.length
+        yield arr[count]
+        count += 1
+      end
+      self
+    end
+    arr.to_enum
+  end
+
+  # my_each_with_index
+  def my_each_with_index(&block)
+    arr = self
+    arr = arr.to_a
+    count = 0
+    if arr.is_a?(Hash)
+      while count < arr.length
+        yield arr[i], count
+        count += 1
+      end
+      self
+    elsif block
+      while count < arr.length
+        yield arr[count], count
+        count += 1
+      end
+      self
+    else
+      arr.to_enum
+    end
+  end
+
+  # my_select
+  def my_select(&block)
+    arr = self
+    return arr.to_enum unless block
+
+    temp = []
+    arr.my_each { |num| temp.push(num) if yield(num) }
+    temp
+  end
+
+  # my_all
+  def my_all?(paramet = nil)
+    an = true
+    arr = self
+    arr = arr.to_a
+    return true if arr.empty?
+
+    if block_given?
+      arr.my_each do |n|
+        ans = yield n
+        return false if ans == false
+      end
+    else
+      an = check_regx(paramet, arr)
+    end
+    an
+  end
+
+  # my_method1
+  def check_regx(paramet, arr)
+    an = flase
+    if paramet.nil?
+      arr.my_each { |n| return false if n.nil? || !n }
+    else
+      an = check_regx_other(paramet, arr)
+    end
+    an
+  end
+
+  def check_regx_other(paramet, arr)
+    if paramet.is_a?(Regexp)
+      arr.my_each { |x| return false unless x.match?(paramet) }
+    elsif paramet.is_a?(Class)
+      arr.my_each { |x| return false unless x.is_a?(paramet) }
+    else
+      arr.my_each { |x| return false if x != paramet }
+    end
+    true
+  end
+
+  # my_any
+  def my_any?(paramet = nil)
+    an = false
+    arr = self
+    arr = arr.to_a
+    return true if arr.empty?
+
+    if block_given?
+      arr.my_each do |n|
+        ans = yield n
+        return true if ans == true
+      end
+    else
+      an = check_other(paramet, arr)
+    end
+    an
+  end
+
+  # my_method2
+  def check_other(paramet, arr)
+    an = false
+    if paramet.nil?
+      arr.my_each { |n| return true if !n == false && !n.nil? }
+    else
+      an = check_other_helper(paramet, arr)
+    end
+    an
+  end
+
+  def check_other_helper(paramet, arr)
+    if paramet.is_a?(Regexp)
+      arr.my_each { |x| return true if x.match?(paramet) }
+    elsif paramet.is_a?(Class)
+      arr.my_each { |x| return true if x.is_a?(paramet) }
+    else
+      arr.my_each { |x| return true if x == paramet }
+    end
+    false
+  end
+
+  # my_none
+  def my_none?(paramet = nil)
+    an = true
+    arr = self
+    arr = arr.to_a
+    return true if arr.empty?
+
+    if block_given?
+      arr.my_each do |n|
+        ans = yield n
+        return false if ans == true
+      end
+    else
+      an = check_class(paramet, arr)
+    end
+    an
+  end
```

Comment:
`my_none?` method, when no block or argument is given returns true only if none of the collection members is true. Please fix it!